lib/restAPIs/library.js

```diff
@@ -0,0 +1,108 @@
+
+module.exports = class RestTopic extends RestUtil {
+  /**
+   * Get base endpoint for this API category
+   */
+  _getBase() {
+    return 'library';
+  }
+
+  /**
+   * Return object representing the library for the authenticated user.
+   * @param {String} outputFlag - Filtered output. DEAFULT = include everything, FILTER_TOC = filter out cahpters and pages.
+   * @param {String} fields - Fields to be included in the result
+   * @return {Object} Library for the authenticated user
+   */
+  getLibrary(outputFlag = 'DEFAULT', fields = '') {
+    const queryParameter = buildCommonQuery(fields);
+    queryParameter.outputFlag = outputFlag;
+
+    return this._makeRequest(
+      this._getBase(),
+      queryParameter,
+      'GET'
+    ).then(result => this.throwIfFailed(result, 200));
+  }
+
+  /**
+   * Publish an object (document or dossier) in project defined by projectID.
+   * Information about object to publish and recipient are detailed in object 'body'.
+   * https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#/Library/publishObject
+   * @param {String} projectID -
+   * @param {object} body
+   */
+  publishObject(publishInfo, projectID) {
+    assert(publishInfo, 'No publishInfo provided');
+
+    const customHeaders = this.getProjectHeader(projectID);
+    return this._makeRequest(
+      this._getBase(),
+      publishInfo,
+      'POST',
+      customHeaders
+    ).then(result => this.throwIfFailed(result, 204));
+  }
+
+  /**
+   * Return object (document or dossier) identified by 'objectId' from project identified by 'projectId'
+   * @param {String} projectId
+   * @param {String} documentId
+   * @param {String} fields - Fields to be included in the result. i.e. 'id,elements'
+   * @returns {Object} - Representing the object
+   */
+  getObject(objectId, projectId, fields = '') {
+    const queryParameter = buildCommonQuery(fields);
+    const endpoint = `${this._getBase()}/${objectId}`;
+    const customHeaders = this.getProjectHeader(projectId);
+    return this._makeRequest(
+      endpoint,
+      queryParameter,
+      'GET',
+      customHeaders
+    ).then(result => this.throwIfFailed(result, 200));
+  }
+
+  /**
+   * Delete an object (dossier or document) identified by 'objectId' from project identified by 'projectId'
+   * @param {String} projectId
+   * @param {String} objectId
+   */
+  deleteObject(objectId, projectId) {
+    const endpoint = `${this._getBase()}/${objectId}`;
+    const queryParameter = buildCommonQuery();
+    const customHeaders = this.getProjectHeader(projectId);
+
+    return this._makeRequest(
+      endpoint,
+      queryParameter,
+      'DELETE',
+      customHeaders
+    ).then(result => this.throwIfFailed(result, 204));
+  }
+
+  /**
+   * Delete object (dossier or document) identified by 'objectId' from project identified by 'projectId' of user identified by 'userId'
+   * @param {String} projectId
+   * @param {Strin} objectId
+   * @param {String} userId
+   */
```

Comment:
typo in type. primitive types here too

Reply:
also, as before, I would make projectID the last parameter - allowing it to be optional if set on a higher level

Reply:
Done.